pinot-core/src/test/java/org/apache/pinot/core/segment/index/creator/SegmentGenerationWithTimeColumnTest.java

```diff
@@ -81,12 +84,25 @@ public void testSimpleDateSegmentGeneration()
   public void testEpochDateSegmentGeneration()
       throws Exception {
     Schema schema = createSchema(false);
-    File segmentDir = buildSegment(schema, false);
+    File segmentDir = buildSegment(schema, false, false);
     SegmentMetadataImpl metadata = SegmentDirectory.loadSegmentMetadata(segmentDir);
     Assert.assertEquals(metadata.getStartTime(), minTime);
     Assert.assertEquals(metadata.getEndTime(), maxTime);
   }
 
+  @Test
+  public void testSegmentGenerationWithInvalidTime() {
+    boolean error = false;
+    try {
+      Schema schema = createSchema(false);
+      buildSegment(schema, false, true);
```

Comment:
You can add a statement below this line like:
```
Assert.fail("...");
```
so that it will guarantee to fail the test if `buildSegment()` passes.

Reply:
Yes, we could do that. However, it won't let me test for the existence of message - exception detail. Right?

Reply:
I think it's ok not to have assertion on the content of the exception; if the exception message changes in the future we'll still need to modify all the checks here. You can just write the code like this:
```
try {
  blabla();
  Assert.fail("...");
} catch (Exception e) {
  // Expected.
}
```

Reply:
Done.